angular_components/lib/simple_html/README.md

```diff
@@ -0,0 +1,297 @@
+            set.
+*   `<simple-html-block>` supports all of the above and the following safe block
+    elements:
+    *   Unordered lists: `<ul>` and `<li>`.
+    *   Paragraphs: `<p>`.
+
+### URI whitelisting {#uri-whitelisting}
+
+By default, only links with the
+[same origin](https://en.wikipedia.org/wiki/Same-origin_policy) as the host page
+as permitted.
+
+Additional paths can be permitted through an optional Angular dependency. See
+the
+[simpleHtmlUriWhitelist](https://github.com/dart-lang/angular_components/blob/master/lib/simple_html/simple_html.dart)
+token for details.
+
+### Custom click handling
+
+The component provides a `(trigger)` output, which is fired whenever an `<a>`
+element with the CSS class `trigger` is clicked by the user. This is intended
+for introducing custom click handlers to specified portions of the text. A
+simple example, firing an Analytics event when a certain word is clicked, might
+define a message and handler in the component:
+
+```dart
+String get someMessage => 'Click <a class="trigger">here</a>.';
+
+void recordAnalyticsEvent => _analytics.fire();
+```
+
+and then include it in the template:
+
+```html
+<simple-html [contents]="someMessage"
+             (trigger)="recordAnalyticsEvent()"></simple-html>
+```
+
+By default, the normal behaviour of the link also occurs, such as navigation. So
+if a user clicks "here" in this message
+
+```html
+Click <a href="//example.com" class="trigger" (trigger)="myTrigger()">here</a>
+```
+
+then they will navigate to example.com _and_ `myTrigger` will be executed.
+
+The trigger definition can also refer to the original Angular
+[`$event`](https://webdev.dartlang.org/angular/guide/template-syntax#event-and-event-handling-statements).
+This can be useful for more advanced features such as:
+
+*   Selectively preventing the default behaviour (such as navigation) by calling
+    [`$event.preventDefault`](https://developer.mozilla.org/en-US/docs/Web/API/Event/preventDefault).
+*   Having multiple `trigger` elements with different behaviour in the same
+    message. Which element was clicked can be identified by reading the
+    [`$event.target`](https://developer.mozilla.org/en-US/docs/Web/API/Event/target)
+    field.
+
+### Limitations
+
+The following is not yet supported:
+
+*   Other elements and attributes that are probably safe but are not yet in
+    demand (such as `blockquote`).
+*   More dangerous cases (such as setting the URL of an iframe) must still
+    be done using `bypassSecurity*` methods. (But preferably don't do this at
+    all.)
+
+## Migration
+
+Migration from an existing `bypassSecurityTrustHtml` solution is generally
+straightforward. Basic steps:
+
+1.  Import the Dart library
+    `package:angular_components/simple_html/simple_html.dart` in your component
+    file.
+1.  Add the `SimpleHtmlComponent` to your
+    [@Component](https://github.com/angular/angular/blob/master/packages/core/src/metadata/directives.ts#L101)'s
```

Comment:
Did you mean to link to an AngularTS source file? :)  Consider using: https://pub.dartlang.org/documentation/angular/latest/angular/Component-class.html

Reply:
Definitely needs to be updated.